tkltest/generate/unit/augment.py

```diff
@@ -387,7 +393,11 @@ def __compute_tests_with_coverage_gain(test_class_augment_pool, ctd_test_dir, ba
             else:
                 logging.info('No coverage gain from test class {}'.format(test_class))
         except subprocess.CalledProcessError as e:
-            logging.error('Error running augmented test suite with class {}: {}'.format(test_class, e))
+            logging.error('Warning: error occurred while merging augmented test suite with class {}: {}'.format(test_class, e))
+
+        if not coverage_delta and not total_coverage:
+            tkltest_status('Terminating augmentation due to memory exhaustion')
```

Comment:
@rachelt44: Is out-of-memory error the only possible root cause for `subprocess.CalledProcessError`? Shouldn't we check for the error type and will all error types be handled in the same way (terminating the augmentation step)?

Reply:
@sinha108 the other error types are also checked but their handling is different that the jacoco merge fail. In the other cases (e.g., failing to execute a specific test), the current test file is skipped and augmentation proceeds to the next test file. Only jacoco merge failure results in terminating augmentation, because when it happens, it will also happen in all subsequent merge attempts.